src/impl.cpp

```diff
@@ -1 +1,33 @@
 #include "impl.h"
+#include <exception>
+#include <stdexcept>
+
+Integer::Integer(int value):_value(value)
+{
+    if ((value>=0) && (value<10))
+    {    std::vector<char*> literalNumbers {"zero",
+                                       "one",
+                                       "two",
+                                       "three",
+                                       "four",
+                                       "five",
+                                       "six",
+                                       "seven",
+                                       "eight",
+                                       "nine"};
+        _buffer = const_cast<char*> (literalNumbers[value]);
```

Comment:
**buffer pointer will be invalidated as soon as literalNumbers will be out of scope!**
literals have a static duration storage, but make this explicit please, mark literalNumbers as static local variable